### ISSUE TITLE
Fix `no-duplicate-selectors` false positives with Less syntax

### ISSUE THREAD

**Issue by FritsvanCampen** [stylelint/stylelint#7887]
### What minimal example or steps are needed to reproduce the bug?

```less
.foo() {
	.bar > & {
		color: #000;
	}
	
	&.readonly {
		color: #fff;
	}
}
```
Note: it's important that the root rule is a mixin (needs brackets)

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "no-duplicate-selectors": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAUBKABMAOgHb4AuSARgIYBOOAfDgGR5GmmQA2E18OAxAAYhAblYkAvmLGMk1GJSgRCHAJ4tiJdhC49+KfaI2TC4kABoQKAJYcYAOUoBbOIhgAPJwAdbSWwGc-c3AlawBzBBACQhwcfBB3EhhCKD843gBtIhiYuL8SFVsOK0ISAFpIQjDSvMpkmig4rJwAXSJTCwqwgDFuR0oSCIArPyUg2E9AxCickDyCmCKStNiQDn6YPLizJtz8wuKyzqtQ6pJaqHrluLXEzZA2kHEgA

### Which Stylelint-related dependencies are you using?

16.8.0

### What did you expect to happen?

No error

### What actually happened?

Stylelint: Unexpected duplicate selector "&.readonly", first used at line 2 (no-duplicate-selectors)

For larger files the errors become non-sensical, so I expect there might be some contextual error.

### Do you have a proposal to fix the bug?

I suspect some logic was changed where less-tokens (such as `&` and `+`) are now incorrectly detected as selectors.

**Comment by romainmenke** [comment]
Thank you for reporting this @FritsvanCampen 

This regression was likely introduced in https://github.com/stylelint/stylelint/pull/7867

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by FritsvanCampen** [comment]
Thanks for the quick fix! 🙏
